src/kernel/mempool_persist.cpp

```diff
@@ -34,14 +34,15 @@ using fsbridge::FopenFn;
 
 namespace kernel {
 
-static const uint64_t MEMPOOL_DUMP_VERSION = 1;
+static const uint64_t MEMPOOL_DUMP_VERSION{2};
 
```

Comment:
not less code, just a bit dry.

Reply:
nit: I agree with @ismaelsadeeq about defining the lambda only once at the top, instead of currently twice (at [`LoadMempool`](https://github.com/bitcoin/bitcoin/blob/eeee4a5779ea20d859cff2e411ad46dd52384f1d/src/kernel/mempool_persist.cpp#L45) and at [`DumpMempool`](https://github.com/bitcoin/bitcoin/blob/eeee4a5779ea20d859cff2e411ad46dd52384f1d/src/kernel/mempool_persist.cpp#L167-L169), the downside is that you have to pass all the arguments on all 4 calls (`CAutoFile file{xor_open(...)};`), other than that I think there's no undesirable behaviour on removing the `&`.

Reply:
Yes resolve, will review soon ✅